sound/soc/sof/ipc4-topology.c

```diff
@@ -993,6 +993,59 @@ static int sof_ipc4_widget_assign_instance_id(struct snd_sof_dev *sdev,
 	return 0;
 }
 
+static void sof_update_hw_param(struct snd_pcm_hw_params *params,
+				snd_pcm_hw_param_t var, unsigned int val)
+{
+	struct snd_mask *m;
+	struct snd_interval *i;
+
+	if (hw_is_mask(var)) {
+		m = hw_param_mask(params, var);
+		snd_mask_none(m);
+		snd_mask_set_format(m, (snd_pcm_format_t)val);
+		return;
+	}
+
+	/* if params is not snd_mask, then it is snd_interval */
```

Comment:
I dont quite understand what this statemeant means "if params is not snd_mask, then it is snd_interval". "params" is obviously not snd_mask. it is snd_pcm_ha_params.